### PR TITLE
Adding developer guide

### PR DESCRIPTION
**Pull request by henriquemoraeszup**
**- What I did**
* Added a developer guide to explain the project structure.
* Edited the PR template to a more open-ended question like **description** instead of **what I did**

**- Description for the changelog**

Added developer guide

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Developer guide
```
Comment (GuillaumeFalourd) [inline DEVELOPER_GUIDE.md:1]: I would add a step by step explaining how to build the project and execute a command locally to test new implementations.
What do you think ?

**Review by GuillaumeFalourd — CHANGES_REQUESTED** [review]

**Comment by GuillaumeFalourd** [comment]
Add a link to the Developer Guide from the README could be interesting as well.

**Comment by Harirai** [comment]
I think this will be very helpful for new contributors @henriquemoraeszup Thank you :tada:  for adding this. I have a few suggestions, I hope they'll be apt. :smile: 

Recently while working on the project, I realized that a description on "How to build the project after making changes", would be very helpful for new developers to use the feature they added/updated. (using `make` command and  purpose of Makefile)

Another nice addition to the guide can be "How to run tests locally", for this there is `run-test.sh` file.

**Comment by henriquemoraeszup** [comment]
Thanks guys for the feedback, after some more meetings with the community I have realised there is still some way to go

**Review by GuillaumeFalourd — APPROVED** [review]

**Pull request merged**